import_ldraw.py

```diff
@@ -928,7 +1004,49 @@ def getColorValue(line, value):
     if value in line:
         n = line.index(value)
         return line[n + 1]
-
+        
+def linkedParts():
+    """Clean-up design by linking identical parts (mesh/color)."""
+    # Generate list of all materials (colors).
+    colors = [color.name for color in bpy.data.materials]
+    
+    # List all unique meshes; for example 3002 and 3002.001 are considered equal.
+    parts = []
+    for part in bpy.data.objects:
+        # Find all unique names of meshes, ignoring everything behind the '.'
+        # and create a list of these names, making sure no double enties occur.
+        if part.type == 'MESH' and part.name.split('.')[0] not in parts:
+            parts.append(part.name.split('.')[0])
+    
+    # For each mesh/color combination create a link to a unique mesh.
+    for part in parts:
+        for color in colors:
+            replaceParts(part, color)
+
+def replaceParts(part, color):
+    """Replace identical meshes of part/color-combination with a linked version."""   
+    scene = bpy.context.scene
+    mat = bpy.data.materials[color]
+    mesh = None
+    
+    # For each object in the scene check if it matches the given part name.
+    for ob in scene.objects:
```

Comment:
Change this line to simply read
```python
for ob in objects:
```
As well as change the comment to start out "For each object we imported check..."